File: epi_judge_java/epi/SearchFirstKey.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

public class SearchFirstKey {

    @EpiTest(testDataFile = "search_first_key.tsv")
    public static int searchFirstOfK(List<Integer> sortedNums, int k) {
        if (sortedNums.isEmpty() || sortedNums.get(sortedNums.size() - 1) < k) return -1;
        int left = -1;
        int right = sortedNums.size();
        while (left + 1 < right) {
            int mid = (left + right) / 2;
            if (sortedNums.get(mid) >= k) right = mid;
            else left = mid;
        }
        return (sortedNums.get(right) == k ? right : -1);
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "SearchFirstKey.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/search_first_key.cc
#include <vector>

#include "test_framework/generic_test.h"

using std::vector;

int SearchFirstOfK(const vector<int> &nums, int k) {
    if (nums.empty()) return -1;
    int left = -1;
    int right = nums.size() - 1;
    while (left + 1 < right) {
        int mid = (left + right) / 2;
        if (nums[mid] < k) {
            left = mid;
        } else {
            right = mid;
        }
    }
    if (nums[right] == k) return right;
    else return -1;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"A", "k"};
    return GenericTestMain(args, "search_first_key.cc", "search_first_key.tsv",
                           &SearchFirstOfK, DefaultComparator{}, param_names);
}
